### DISCUSSION TITLE
cssrlib-test_hasppp-Mild Warning -- time crossed leap second table  boundaries

### DISCUSSION THREAD

**Discussion by yiheli2022** [hirokawa/cssrlib#57] [Q&A]
Hi,

   I collected  some data with sepetentrio mosaci (rinex 3.04 and galrawcnav.txt) see attachment. I tried the sample data with test_hasppp.py, it worked fine. But when I tried with my data. It shows Mild Warning -- time crossed leap second table
   boundaries.  Boundary edge value used instead and some other unnormal logs. Can you help me find the reason ?

Best regards,
Yihe 

liyihe_2005@163.com
[Uploading 20240807_galrawcnav.txt…]()

**Comment by yiheli2022** [comment]
[20240807_galrawcnav.zip](https://github.com/user-attachments/files/16550567/20240807_galrawcnav.zip)
see data attached

**Reply by AndreHauschild** [reply]
Hello Yihe,

thanks for your question and my apologies for the long delay in getting back to you. I hope my answers below are still valuable to you.

Firstly, the warnings you are seeing originate from the PySolid module, which is used for the computation of the site displacements. I found out that it issues this warning when a validity interval for the leap-second table is exceeded, which is currently set to June 28, 2024. Your dataset from August 8 may be the most recent processed with cssrlib so far, that is why nobody has noticed until now. Thanks for pointing this out! I guess that the developers of PySolid will update the validity interval eventually and the warning will disappear. Until then, you can safely ignore it, since no new leap seconds have been introduced and the table in PySolid is still up to date.

Looking into your RINEX files I noticed two other problems:

1. The antenna type is not defined in the file header. This is required to compute the antenna corrections, so adding this information will help.
2. There are no C/N0 observations provided. Observations with low C/N0 are rejected from the processing, so this will be useful to have. Maybe you can generate your RINEX files again with the C/N0 included. Otherwise you have to set the corresponding rejection threshold to zero. You can do this in test_ppphas.py by defining the new limit after the initialization of the PPP processing like this:

```
ppp = pppos(nav, rnx.pos, 'test_ppphas.log').
nav.cnr_min = nav.cnr_min_gpy = 0
```
I hope that helps!

Kind regards, André
